src/security/builtin_plugins/authentication/src/authentication.c

```diff
@@ -1338,14 +1364,14 @@ static X509 *load_X509_certificate_from_binprop (const DDS_Security_BinaryProper
 
   DDS_Security_ValidationResult_t result = DDS_SECURITY_VALIDATION_FAILED;
   if (trusted_ca_list->length == 0)
-    result = verify_certificate (cert, own_ca, ex);
+    result = verify_certificate (cert, own_ca, own_crl, ex);
   else
   {
     DDS_Security_Exception_clean (ex);
     for (unsigned i = 0; i < trusted_ca_list->length; ++i)
     {
       DDS_Security_Exception_reset (ex);
-      if ((result = verify_certificate (cert, trusted_ca_list->buffer[i], ex)) == DDS_SECURITY_VALIDATION_OK)
+      if ((result = verify_certificate (cert, trusted_ca_list->buffer[i], NULL, ex)) == DDS_SECURITY_VALIDATION_OK)
```

Comment:
> It seems strange to not apply the CRL if there is a list of trusted CAs, that means that a small change in the way the CA is configured could cause the CRL to be ignored. Perhaps for this it would be more elegant to allow a set of CRLs (like the it allows a set of CAs), but that is much more work for very little gain.

Yeah, we could allow that, but it seemed overkill to me.  There's always room in the future to define a new option for multiple of these if we really want.

> If it is unwise to apply one CRL for different CAs, then I would just fail the check if a CRL is configured and there are multiple CAs. At least that is fail-closed.

As far as I understand, the CRL is signed by a particular CA, and so passing it through to the other set of trusted CAs would result in failures to decrypt it properly.  In that sense, the property is really "crl-from-the-root-CA".  But I like your idea of failing the check if both CRL and trusted_ca_list are both provided.  I'll go ahead and implement that instead.